src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs

```diff
@@ -211,24 +257,31 @@ public static bool IsLockFileStillValid(DependencyGraphSpec dgSpec, PackagesLock
                                     }
                                     else // This should never happen.
                                     {
-                                        return false;
+                                        return (false, string.Empty);
```

Comment:
I wonder if we should add a message that says "Hey, if you see this, please file an issue on the product :D )

Reply:
Modified code to throw exception in this case as per Andy's code review feedback. Pls review again and let me know the feedback.